nion/swift/test/ImageCanvasItem_test.py

```diff
@@ -291,6 +292,27 @@ def test_hand_tool_on_one_image_of_multiple_displays(self):
             document_controller.tool_mode = "hand"
             display_panel.display_canvas_item.simulate_press((100,125))
 
+    def test_zoom_tool_in_and_out_around_clicked_point(self):
+        # setup
+        with TestContext.create_memory_context() as test_context:
+            document_controller = test_context.create_document_controller()
+            document_model = document_controller.document_model
+            display_panel = document_controller.selected_display_panel
+            data_item = DataItem.DataItem(numpy.zeros((10, 10)))
+            document_model.append_data_item(data_item)
+            display_item = document_model.get_display_item_for_data_item(data_item)
+            copy_display_item = document_model.get_display_item_copy_new(display_item)
+            display_panel.set_display_panel_display_item(copy_display_item)
+            header_height = display_panel.header_canvas_item.header_height
+            display_panel.root_container.layout_immediate((1000 + header_height, 1000))
+            # run test
+            document_controller.tool_mode = "zoom"
+            display_panel.display_canvas_item.simulate_press((100, 125))
+            display_panel.display_canvas_item.simulate_release((100, 125))
+
```

Comment:
What's missing from this test is confirmation that the zoom tool is performing as expected, i.e. by checking image position and image zoom. In addition, I'd like to see a test with the other view modes as initial conditions (i.e. 1:1 mode, etc.). A quick test of the functionality reveals to me some problems that should be caught by (a) manual testing and (b) automated tests here:

- zoom out gets way off center after about 4 zoom outs
- starting with 1:1 zooming (press '1' in panel to get to that state) or 'fill' zoom mode (press '0') do not give expected results.

See also this issue, which may need to go in before this PR:
- #1169